### PR TITLE
Generated Typescript App

### PR DESCRIPTION
**Pull request by JRJurman**
## Summary
This PR introduces typescript for the generated app. This allows people to build apps with the types that are provided by Tram-One, and all of the nice tools and utilities that come with Typescript.

## Checklist
- [x] PR Summary
- [x] PR Annotations
- [x] Tests
- [x] Version Bump

### CODE REVIEW

File: template/src/app.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { registerHtml } from "tram-one";
+import { registerHtml, TramOneComponent } from "tram-one";
 import logo from "@tram-one/tram-logo/dist/element";
 import "./app.css";
 
 const html = registerHtml();
 
-export default () => {
+const app: TramOneComponent = () => {
```
Comment (JRJurman) [inline template/src/app.ts:7]: This is honestly the most significant part of the change, we now pull in one of the types included in Tram-One, and type our component with it.

File: template/src/app.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { screen } from "@testing-library/dom";
```
Comment (JRJurman) [inline template/src/app.test.ts:1]: using `screen` now as this is the preference for testing-library react, and reflects the side-effect nature of our tests now.

File: template/src/app.test.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+describe("app", () => {
+  afterEach(() => {
+    document.body.innerHTML = "";
+  });
+
+  it("should render a link to the tram-one site", () => {
+    start(app, "body");
+
+    const link = screen.queryByText("Learn Tram-One");
+    expect(link).toHaveAttribute("href", "https://tram-one.io");
```
Comment (JRJurman) [inline template/src/app.test.ts:14]: changed this test to just start Tram-One with the app as the component. This ensures that all the scaffolding that is required is actually laid out, whereas the previous test code would have missed that.

Comment (JRJurman) [reply template/src/app.test.ts:14]: additionally (and the primary purpose for changing this), calling app on it's own actually caused a type error, first that the component needed two parameters (props and children) and then that it wasn't of type HTMLElement, requiring the following code to suffice
```typescript
const renderedApp = app(undefined, undefined) as unknown as HTMLElement;
```

While thinking about this, I realized that `start` was better for all the other things that could break here, so I think this solution is fine. 

I do need to go back to Tram-One and make those parameters optional and see what we can do to make `TramOneElement` better map to `HTMLElement`.

File: template/.tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "include": ["./src/**/*"],
+  "compilerOptions": {
+    "esModuleInterop": true
+  }
+}
```
Comment (JRJurman) [inline template/.tsconfig.json:6]: this was the most minimal typescript config that I believe will "just work" - mostly to make jest happy.

File: template/src/app.test.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { screen } from "@testing-library/dom";
+import { start } from "tram-one";
+import app from "./app";
+
+describe("app", () => {
```
Comment (chtinahow) [inline template/src/app.test.ts:5]: why double quotes here?

**Review by chtinahow — APPROVED** [review]
Just the question, but nothing blocking. LGTM 👍

File: template/src/app.test.ts (hunk 5 above)
Comment (JRJurman) [reply template/src/app.test.ts:5]: This is a side effect of the minimal prettier config in the project (which defaults to double quotes)

**Pull request merged**